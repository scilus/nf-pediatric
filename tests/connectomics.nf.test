nextflow_pipeline {

    options "-stub-run"

    name "Test nf-pediatric profile connectomics"
    script "../main.nf"

    test("Connectomics profile with derivatives - should run successfully") {

        when {
            params {

                params.input_deriv = "$projectDir/tests/data/derivatives/"
                params.outdir = "$outputDir"

                params.connectomics = true

            }
        }

        then {
            // stable name: All files + folders in ${params.outdir}/ with a stable name.
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore:  ['pipeline_info/*.{html,json,txt}'])
            assertAll {
                assert workflow.success
                assert snapshot(
                    // Number of successfully completed tasks
                    workflow.trace.succeeded().size(),
                    // Remove the nextflow version from the versions.yml because we test it using different nextflow versions.
                    removeNextflowVersion("$outputDir/pipeline_info/nf-pediatric_software_mqc_versions.yml"),
                    // All stable name.
                    stable_name
                ).match()
            }
        }
    }

    test("Connectomics profile with derivatives and session - should run successfully") {

        when {
            params {

                params.input_deriv = "$projectDir/tests/data/derivatives/"
                params.outdir = "$outputDir"

                params.connectomics = true

            }
        }

        then {
            // stable name: All files + folders in ${params.outdir}/ with a stable name.
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore:  ['pipeline_info/*.{html,json,txt}'])
            assertAll {
                assert workflow.success
                assert snapshot(
                    // Number of successfully completed tasks
                    workflow.trace.succeeded().size(),
                    // Remove the nextflow version from the versions.yml because we test it using different nextflow versions.
                    removeNextflowVersion("$outputDir/pipeline_info/nf-pediatric_software_mqc_versions.yml"),
                    // All stable name.
                    stable_name
                ).match()
            }
        }
    }

    test("Connectomics profile with no participants.tsv file - should fail") {

        when {
            params {

                params.input_deriv = "$projectDir/tests/data/noparticipantfile/"
                params.outdir = "$outputDir"

                params.connectomics = true

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("ERROR: Your bids dataset does not contain a participants.tsv file. Please provide a participants.tsv file with a column indicating the participants' age. For any questions, please refer to the documentation at https://github.com/scilus/nf-pediatric.git or open an issue!")
        }
    }

    test("Connectomics profile with wrong participants.tsv file - should fail") {

        when {
            params {

                params.input_deriv = "$projectDir/tests/data/wrongparticipantfile/"
                params.outdir = "$outputDir"

                params.connectomics = true

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("ERROR: Age is not entered correctly in the participants.tsv file. Please validate.")
        }
    }
}
