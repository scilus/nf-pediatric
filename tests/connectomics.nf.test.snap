{
    "Connectomics profile with derivatives and session - should run successfully": {
        "content": [
            35,
            {
                "CONNECTIVITY_AFDFIXEL": {
                    "scilpy": 1.6
                },
                "CONNECTIVITY_DECOMPOSE": {
                    "scilpy": "2.0.2"
                },
                "CONNECTIVITY_METRICS": {
                    "scilpy": "2.0.2"
                },
                "CONNECTIVITY_VISUALIZE": {
                    "scilpy": "2.0.2"
                },
                "FILTERING_COMMIT": {
                    "scilpy": 1.6
                },
                "TRANSFORM_LABELS": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": null
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "dataset_description.json",
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test1",
                "sub-test1/ses-baseline",
                "sub-test1/ses-baseline/anat",
                "sub-test1/ses-baseline/anat/sub-test1_ses-baseline_space-DWI_seg-DK_dseg.nii.gz",
                "sub-test1/ses-baseline/dwi",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_desc-filtered_tractogram.h5",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_desc-preproc_tractogram.h5",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_stat-ad.npy",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_stat-fa.npy",
                "sub-test1/ses-baseline/figures",
                "sub-test1/ses-baseline/figures/sub-test1_gradients_mqc.png",
                "sub-test1/ses-baseline/figures/sub-test1_labels_mqc.png",
                "sub-test1/ses-baseline/figures/sub-test1_ses-baseline_seg-DK_stat-ad.png",
                "sub-test1/ses-baseline/figures/sub-test1_ses-baseline_seg-DK_stat-fa.png",
                "sub-test1/ses-baseline/multiqc",
                "sub-test1/ses-baseline/multiqc/sub-test1_multiqc_data",
                "sub-test1/ses-baseline/multiqc/sub-test1_multiqc_plots",
                "sub-test1/ses-baseline/multiqc/sub-test1_multiqc_report.html",
                "sub-test2",
                "sub-test2/dwi",
                "sub-test2/dwi/sub-test2_seg-BrainnetomeChild_desc-filtered_tractogram.h5",
                "sub-test2/dwi/sub-test2_seg-BrainnetomeChild_desc-preproc_tractogram.h5",
                "sub-test2/dwi/sub-test2_seg-DK_stat-ad.npy",
                "sub-test2/dwi/sub-test2_seg-DK_stat-afd.npy",
                "sub-test2/dwi/sub-test2_seg-DK_stat-fa.npy",
                "sub-test2/dwi/sub-test2_seg-DK_stat-nufo.npy",
                "sub-test2/figures",
                "sub-test2/figures/sub-test2_gradients_mqc.png",
                "sub-test2/figures/sub-test2_labels_mqc.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-ad.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-afd.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-fa.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-nufo.png",
                "sub-test2/multiqc",
                "sub-test2/multiqc/sub-test2_multiqc_data",
                "sub-test2/multiqc/sub-test2_multiqc_plots",
                "sub-test2/multiqc/sub-test2_multiqc_report.html",
                "sub-test3",
                "sub-test3/anat",
                "sub-test3/anat/sub-test3_space-DWI_seg-BrainnetomeChild_dseg.nii.gz",
                "sub-test3/dwi",
                "sub-test3/dwi/sub-test3_seg-BrainnetomeChild_desc-filtered_tractogram.h5",
                "sub-test3/dwi/sub-test3_seg-BrainnetomeChild_desc-preproc_tractogram.h5",
                "sub-test3/dwi/sub-test3_seg-DK_stat-fa.npy",
                "sub-test3/dwi/sub-test3_seg-DK_stat-md.npy",
                "sub-test3/figures",
                "sub-test3/figures/sub-test3_gradients_mqc.png",
                "sub-test3/figures/sub-test3_labels_mqc.png",
                "sub-test3/figures/sub-test3_seg-DK_stat-fa.png",
                "sub-test3/figures/sub-test3_seg-DK_stat-md.png",
                "sub-test3/multiqc",
                "sub-test3/multiqc/sub-test3_multiqc_data",
                "sub-test3/multiqc/sub-test3_multiqc_plots",
                "sub-test3/multiqc/sub-test3_multiqc_report.html",
                "sub-test4",
                "sub-test4/dwi",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_desc-filtered_tractogram.h5",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_desc-preproc_tractogram.h5",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_stat-fa.npy",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_stat-md.npy",
                "sub-test4/figures",
                "sub-test4/figures/sub-test4_gradients_mqc.png",
                "sub-test4/figures/sub-test4_labels_mqc.png",
                "sub-test4/figures/sub-test4_seg-BrainnetomeChild_stat-fa.png",
                "sub-test4/figures/sub-test4_seg-BrainnetomeChild_stat-md.png",
                "sub-test4/multiqc",
                "sub-test4/multiqc/sub-test4_multiqc_data",
                "sub-test4/multiqc/sub-test4_multiqc_plots",
                "sub-test4/multiqc/sub-test4_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-20T21:50:04.615041"
    },
    "Connectomics profile with derivatives - should run successfully": {
        "content": [
            35,
            {
                "CONNECTIVITY_AFDFIXEL": {
                    "scilpy": 1.6
                },
                "CONNECTIVITY_DECOMPOSE": {
                    "scilpy": "2.0.2"
                },
                "CONNECTIVITY_METRICS": {
                    "scilpy": "2.0.2"
                },
                "CONNECTIVITY_VISUALIZE": {
                    "scilpy": "2.0.2"
                },
                "FILTERING_COMMIT": {
                    "scilpy": 1.6
                },
                "TRANSFORM_LABELS": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": null
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "dataset_description.json",
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test1",
                "sub-test1/ses-baseline",
                "sub-test1/ses-baseline/anat",
                "sub-test1/ses-baseline/anat/sub-test1_ses-baseline_space-DWI_seg-DK_dseg.nii.gz",
                "sub-test1/ses-baseline/dwi",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_desc-filtered_tractogram.h5",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_desc-preproc_tractogram.h5",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_stat-ad.npy",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_seg-DK_stat-fa.npy",
                "sub-test1/ses-baseline/figures",
                "sub-test1/ses-baseline/figures/sub-test1_gradients_mqc.png",
                "sub-test1/ses-baseline/figures/sub-test1_labels_mqc.png",
                "sub-test1/ses-baseline/figures/sub-test1_ses-baseline_seg-DK_stat-ad.png",
                "sub-test1/ses-baseline/figures/sub-test1_ses-baseline_seg-DK_stat-fa.png",
                "sub-test1/ses-baseline/multiqc",
                "sub-test1/ses-baseline/multiqc/sub-test1_multiqc_data",
                "sub-test1/ses-baseline/multiqc/sub-test1_multiqc_plots",
                "sub-test1/ses-baseline/multiqc/sub-test1_multiqc_report.html",
                "sub-test2",
                "sub-test2/dwi",
                "sub-test2/dwi/sub-test2_seg-BrainnetomeChild_desc-filtered_tractogram.h5",
                "sub-test2/dwi/sub-test2_seg-BrainnetomeChild_desc-preproc_tractogram.h5",
                "sub-test2/dwi/sub-test2_seg-DK_stat-ad.npy",
                "sub-test2/dwi/sub-test2_seg-DK_stat-afd.npy",
                "sub-test2/dwi/sub-test2_seg-DK_stat-fa.npy",
                "sub-test2/dwi/sub-test2_seg-DK_stat-nufo.npy",
                "sub-test2/figures",
                "sub-test2/figures/sub-test2_gradients_mqc.png",
                "sub-test2/figures/sub-test2_labels_mqc.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-ad.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-afd.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-fa.png",
                "sub-test2/figures/sub-test2_seg-DK_stat-nufo.png",
                "sub-test2/multiqc",
                "sub-test2/multiqc/sub-test2_multiqc_data",
                "sub-test2/multiqc/sub-test2_multiqc_plots",
                "sub-test2/multiqc/sub-test2_multiqc_report.html",
                "sub-test3",
                "sub-test3/anat",
                "sub-test3/anat/sub-test3_space-DWI_seg-BrainnetomeChild_dseg.nii.gz",
                "sub-test3/dwi",
                "sub-test3/dwi/sub-test3_seg-BrainnetomeChild_desc-filtered_tractogram.h5",
                "sub-test3/dwi/sub-test3_seg-BrainnetomeChild_desc-preproc_tractogram.h5",
                "sub-test3/dwi/sub-test3_seg-DK_stat-fa.npy",
                "sub-test3/dwi/sub-test3_seg-DK_stat-md.npy",
                "sub-test3/figures",
                "sub-test3/figures/sub-test3_gradients_mqc.png",
                "sub-test3/figures/sub-test3_labels_mqc.png",
                "sub-test3/figures/sub-test3_seg-DK_stat-fa.png",
                "sub-test3/figures/sub-test3_seg-DK_stat-md.png",
                "sub-test3/multiqc",
                "sub-test3/multiqc/sub-test3_multiqc_data",
                "sub-test3/multiqc/sub-test3_multiqc_plots",
                "sub-test3/multiqc/sub-test3_multiqc_report.html",
                "sub-test4",
                "sub-test4/dwi",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_desc-filtered_tractogram.h5",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_desc-preproc_tractogram.h5",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_stat-fa.npy",
                "sub-test4/dwi/sub-test4_seg-BrainnetomeChild_stat-md.npy",
                "sub-test4/figures",
                "sub-test4/figures/sub-test4_gradients_mqc.png",
                "sub-test4/figures/sub-test4_labels_mqc.png",
                "sub-test4/figures/sub-test4_seg-BrainnetomeChild_stat-fa.png",
                "sub-test4/figures/sub-test4_seg-BrainnetomeChild_stat-md.png",
                "sub-test4/multiqc",
                "sub-test4/multiqc/sub-test4_multiqc_data",
                "sub-test4/multiqc/sub-test4_multiqc_plots",
                "sub-test4/multiqc/sub-test4_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-20T21:48:01.6684"
    }
}